test/unit/transform/classes_test.py

```diff
@@ -909,16 +910,51 @@ def test_drop_invalid_rows(df, expected, params):
 
 
 @pytest.mark.parametrize(
-    "df,expected,params",
-    [pytest.param(MIXED_TYPE_DATA, SPOT_FIXED_MIXED_TYPE_DATA, SPOT_PARAMS)],
+    "df,expected,params,errors",
+    [
+        pytest.param(
+            MIXED_TYPE_DATA, SPOT_FIXED_MIXED_TYPE_DATA, SPOT_PARAMS, does_not_raise()
+        ),
+        pytest.param(
+            MIXED_TYPE_DATA,
+            SPOT_FIXED_MIXED_TYPE_DATA,
```

Comment:
@zaneselvans New tests for failures live here and below.

Reply:
The kind of test case I was imagining to check that the string/numeric type casting works would look something like:

```py
{
    "idx_cols": ["id"],
    "fix_cols": ["plant_name"],
    "expect_unique": True,
    "spot_fixes": [
        (2, 2000.0),  # Output should be a plant with the name "2000.0"
    ],
},
{
    "idx_cols": ["id"],
    "fix_cols": ["year"],
    "expect_unique": True,
    "spot_fixes": [
        (2, "2076"), # Output should have a year that's an integer 2076
    ],
}
```

Reply:
Ah sorry, misread and completely missed that one. I've just added it as well.